client/state/reader/site-blocks/reducer.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	READER_SITE_BLOCK_REQUEST_SUCCESS,
+	READER_SITE_UNBLOCK_REQUEST_SUCCESS,
+	SERIALIZE,
+	DESERIALIZE,
+} from 'state/action-types';
+
+/**
+ * Tracks all known block statuses, indexed by site ID.
+ *
+ * @param  {Array} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Array}        Updated state
+ */
+export function items( state = {}, action ) {
+	switch ( action.type ) {
+		case READER_SITE_BLOCK_REQUEST_SUCCESS:
+			return {
+				...state,
+				[ action.siteId ]: action.data.success
```

Comment:
This is tricky - the endpoint doesn't raise the alarm if there's a problem making the block. It just returns `success: false` :/

For example:

```
{
	"success": false,
	"modified_sub": false
}
```

I think we'll either need a new version of the endpoint that _will_ trigger `READER_SITE_BLOCK_REQUEST_FAILURE` when there's a problem, or we need record an error if `action.data.success` is false.